Readme 2.0

shadycuz opened the pull request.

shadycuz commented on aws/efs/README.md:
````diff
@@ -5,9 +5,109 @@
+        claimName: efs
+```
+
+Because our claim is ReadWriteMany if you scale this pod each aditional pod will also be able to read and write the same files. You may also reference the same claimName in another type of pod so your 2 applications can read and write the same files. If you wish to have a second application that uses EFS storage but don't want other pods to access the files, create a new claim using a new name bu ...
````
Not sure if this is completely accurate.

wongma7 replied:
I would just remove this 'Because our claim is ReadWriteMany...', see below comments

shadycuz commented on aws/efs/README.md:
```diff
@@ -134,3 +235,37 @@ Add a reference to the secret in the deployment yaml.
+
+- Can I omit that part of the claim?
+
+Good question... 
```
Wasn't sure

shadycuz commented:
Oh I and I didn't test my manifest file or some of the assumptions I made. I will test tomorrow.

wongma7 commented on aws/efs/README.md:
```diff
@@ -5,9 +5,109 @@
+
+Because our claim is ReadWriteMany if you scale this pod each aditional pod will also be able to read and write the same files. You may also reference the same claimName in another type of pod so your 2 applications can read and write the same files. If you wish to have a second application that uses EFS storage but don't want other pods to access the files, create a new claim using a new name bu ...
+
+Some times you want the replica pods to be on EFS but you do not wish them to share the same files. In those situations it's best to use a [StatefulSet](./https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/) with the setting ReadWriteOnce. When a StatefulSet scales up it will dynamicly create new claims for your pods.
```
RWO and RWM are just labels, they don't actually enforce anything at the moment. Even if you ask for RWO, the RWO PV can still be read/written to from multiple pods if multiple pods consume the PVC, kubernetes won't care.

wongma7 replied:
For that reason I would avoid mentioning RWO and RWM in the line above, it is just confusing IMO. EFS is NFS which has the property of being RWM, I would leave it at that

wongma7 commented on aws/efs/README.md:
````diff
@@ -5,9 +5,109 @@
+```
+Note: We do not reference a claim name, instead we give the new claim the name efs and we list the StorageClass we wish to use when creating the claim. I like to use the type of storage, because the name of the pod will be be added to it. This example when ran will create the claim `efs-web-0` and scaling the pod up to 3 will create `efs-web-1` and `efs-web-2`.
+
+If you wish to learn more about efs-provisioner and how to change additional settings you contine on in the deployment section. You might also want to check the FAQ at the bottom.
````
...settings, continue on...

wongma7 commented on aws/efs/README.md:
````diff
@@ -5,9 +5,109 @@
 quay.io/external_storage/efs-provisioner:v0.1.0
 ```
 
+The efs-provisioner is a storage plugin for kubernetes. It consists of a container that has access to an AWS [EFS](https://aws.amazon.com/efs/) resource. The container reads a configmap which contains the EFS filesystem ID, the AWS region and the name you want to use for your efs-provisioner. This name will be used later when you create a storage class.
````
storage plugin isn't right, at best this can be called a subset of a storage plugin. storage plugin implies a full flex volume plugin or container storage interface implementation which are much more than this.

shadycuz commented:
@wongma7 I will make the changes as soon as I can.

shadycuz commented:
@wongma7 I have fixed the issues you found and have changed the manifest and updated the getting started section to no longer need an existing directory thanks to [#261(comment) ](https://github.com/kubernetes-incubator/external-storage/issues/241#issuecomment-317336524).
